### PR TITLE
Add a reset() method to nvFuser FusionCache to enable proper resetting during tests.

### PR DESCRIPTION
**Pull request by kevinstephano**
Fixes issue Jie found in his PR:

https://github.com/pytorch/pytorch/pull/84626#issuecomment-1250745334

### CODE REVIEW

File: torch/csrc/jit/codegen/cuda/python_frontend/fusion_cache.cpp [hunk 1]
```diff
@@ -19,6 +19,7 @@ FusionCache* FusionCache::get(size_t max_fusions) {
   if (singleton_ == nullptr) {
     singleton_ = new FusionCache(max_fusions);
   }
+  singleton_->max_fusions_ = max_fusions;
```
Comment (jjsjann123) [inline torch/csrc/jit/codegen/cuda/python_frontend/fusion_cache.cpp:22]: Should we check `fusions_.size() <= max_fuions`?

**Review by jjsjann123 — APPROVED** [review]
LGTM, minor comments

File: torch/csrc/jit/codegen/cuda/python_frontend/fusion_cache.cpp (hunk 1 above)
Comment (kevinstephano) [reply torch/csrc/jit/codegen/cuda/python_frontend/fusion_cache.cpp:24]: Yes, that is a good idea.  I added the check.

**Pull request closed without merging**